### PR TITLE
fix for banks(crypt, ships) not giving morale penalty

### PR DESCRIPTION
**Pull request by godric3**
also texts should be closer to original now: on visit we can decide if we want to search bank, if already cleared some of them give morale penalty

fix https://bugs.vcmi.eu/view.php?id=2756

### CODE REVIEW

File: lib/mapObjects/CBank.cpp [hunk 1]
```diff
@@ -262,32 +265,35 @@ void CBank::doVisit(const CGHeroInstance * hero) const
 			std::set<SpellID> spells;
 
 			bool noWisdom = false;
+			if(textID == 106){
```
Comment (alexvins) [inline lib/mapObjects/CBank.cpp]: Put `{` on new line

File: lib/mapObjects/CBank.cpp [hunk 2]
```diff
@@ -95,124 +95,127 @@ bool CBank::wasVisited (PlayerColor player) const
...
-		break;
-	case Obj::PYRAMID:
-		textID = 106;
+		case Obj::CREATURE_BANK:
+		case Obj::DRAGON_UTOPIA:
+		default:
+			iw.text << VLC->generaltexth->advobtxt[33];// This was X, now is completely empty
+			iw.text.addReplacement(getObjectName());
+		}
+		if (textID != -1) {
```
Comment (alexvins) [inline lib/mapObjects/CBank.cpp]: Put `{` on new line

File: lib/mapObjects/CBank.cpp [hunk 3]
```diff
@@ -95,124 +95,127 @@ bool CBank::wasVisited (PlayerColor player) const
...
-		if (bc)
-			textID = 124;
-		else
-			textID = 123;
-		break;
-	case Obj::PYRAMID:
-		textID = 106;
+		case Obj::CREATURE_BANK:
+		case Obj::DRAGON_UTOPIA:
+		default:
```
Comment (alexvins) [inline lib/mapObjects/CBank.cpp]: What is the point of such case label? Should it better to be just `default`?

Comment (godric3) [reply lib/mapObjects/CBank.cpp]: Just readability, searching for creature bank, or utopia code will be easier with such case.
Not sure if default is needed, can it occur?

Comment (alexvins) [reply lib/mapObjects/CBank.cpp]: (Modded) Creature bank can have any ID, it defined by `handler` field.

Comment (godric3) [reply lib/mapObjects/CBank.cpp]: So shouldn't `default` be added to two other switch statements, with some generic text? lines 102 and 142

Comment (alexvins) [reply lib/mapObjects/CBank.cpp]: I think it should

Comment (godric3) [reply lib/mapObjects/CBank.cpp]: fixed braces, and` default`

File: lib/mapObjects/CBank.cpp [hunk 4]
```diff
@@ -95,124 +95,130 @@ bool CBank::wasVisited (PlayerColor player) const
...
+	default:
+		banktext = 32;
+		break;
+	}
+	BlockingDialog bd(true, false);
+	bd.player = h->getOwner();
+	bd.soundID = soundID;
+	bd.text.addTxt(MetaString::ADVOB_TXT, banktext);
+	if (ID == Obj::CREATURE_BANK)
+		bd.text.addReplacement(getObjectName());
```
Comment (alexvins) [inline lib/mapObjects/CBank.cpp]: default case also use text 32 and should use object name replacement

Comment (godric3) [reply lib/mapObjects/CBank.cpp]: fixed

**Pull request merged**